GitUp/Application/CloneWindowController.m

```diff
@@ -0,0 +1,109 @@
+//
+//  CloneWindowController.m
+//  Application
+//
+//  Created by Dmitry Lobanov on 08.10.2019.
+//
+
+#import "CloneWindowController.h"
+#import <GitUpKit/GitUpKit.h>
+
+@interface CloneWindowControllerResult ()
+- (instancetype)initWithRepositoryURL:(NSURL *)url directoryPath:(NSString *)path recursive:(BOOL)recursive;
+@end
+
+@implementation CloneWindowControllerResult
+- (instancetype)initWithRepositoryURL:(NSURL *)url directoryPath:(NSString *)path recursive:(BOOL)recursive {  
+  if ((self = [super init])) {
+    self.repositoryURL = url;
+    self.directoryPath = path;
+    self.recursive = recursive;
+  }
+  return self;
+}
+
+- (BOOL)invalidRepository {
+  return self.repositoryURL == nil;
+}
+
+- (BOOL)emptyDirectoryPath {
+  return self.directoryPath == nil;
+}
+@end
+
+@interface CloneWindowController ()
+@property(nonatomic, weak) IBOutlet NSTextField* urlTextField;
+@property(nonatomic, weak) IBOutlet NSButton* cloneRecursiveButton;
+@property (nonatomic, assign, readonly) BOOL urlExists;
+@property (nonatomic, assign, readonly) BOOL recursive;
```

Comment:
`assign` is redundant for `BOOL` type. Just remove it.

Reply:
@lucasderraugh Fixed!